refactor: echo 27: Introducing biome

luarmr opened the pull request:
Introducing biome. 
First run and manual changes.

luarmr committed:
add hook

adding pre-commit hook

update hook to 1.7.0

luarmr committed:
excluding 2 random errors, follow up

luarmr committed:
adding biome to ci

juliosgarbi commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8755198929)
> Docker image was pushed with the tag `20240419.142923-fb-echo-27-regenerate-fir-88feb23cf`

farioas commented on .github/workflows/biome.yml:
```diff
@@ -0,0 +1,43 @@
+        shell: bash
+        run: |
+          set -xeuo pipefail
+          git config --global user.name 'robot-ci-heartex'
```
Will it push changes into branch back?
If no - it's not required

luarmr replied:
I try to stick to the pattern I found in the platform, although I may have chosen the wrong one. Thank you for the review. This PR is mostly generated through a script and I will re-run the script on the day of the release to ensure that biome is applied to any new changes. 

Should I separate this for this PR and create another one, where biome is not required, but already in the platform? And in this PR, should I only do the required part?

luarmr replied:
> Will it push changes into branch back?

My only intention is to block the pr not alter the the branch
So i thought this will need it to pull the branch, so I don't need the complete Configure git?

luarmr replied:
Done!

farioas commented on .github/workflows/biome.yml:
```diff
@@ -0,0 +1,43 @@
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
```
inputs.head_sha

luarmr replied:
Should we review this in other workflows?

nikitabelonogov committed:
fix input var

luarmr committed:
Remove unecesary github conf set up on action ci

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8805867596)
> Successfully merged:  1 file changed, 1 insertion(+), 1 deletion(-)

juliosgarbi commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8817884812)
> Docker image was pushed with the tag `20240424.140531-fb-echo-27-regenerate-fir-d0f102949`

luarmr commented:
Changed to draft to ensure it doesn't merge on its own before Monday.

yyassi-heartex reviewed (approved):
approved but noting that there is are merge conflicts that need to be addressed

luarmr committed:
refactor: ECHO-27: introducting biome, removing remining eslint

luarmr committed:
first biome configuration excluding all unsafe and safe that are not corrected automaticly

luarmr committed:
Apply biome check corrections

luarmr committed:
Apply biome check corrections second passs

luarmr committed:
noAriaHiddenOnFocusable

luarmr committed:
noAriaUnsupportedElements

luarmr committed:
noDistractingElements

luarmr committed:
noInteractiveElementToNoninteractiveRole

luarmr committed:
noHeaderScope

luarmr committed:
noNoninteractiveTabindex

luarmr committed:
noNoninteractiveElementToInteractiveRole

luarmr committed:
noRedundantRoles

luarmr committed:
noPositiveTabindex

luarmr committed:
useAriaActivedescendantWithTabindex

luarmr committed:
useAnchorContent

luarmr committed:
useValidAriaProps

luarmr committed:
useValidAriaRole

luarmr committed:
noExtraBooleanCast

luarmr committed:
noThisInStatic

luarmr committed:
noUselessLoneBlockStatements

luarmr committed:
noUselessConstructor

luarmr committed:
noUselessTernary

luarmr committed:
useLiteralKeys

luarmr committed:
useRegexLiterals

luarmr committed:
noConstAssign

luarmr committed:
useSimplifiedLogicExpression

luarmr committed:
noNewSymbol

luarmr committed:
noInvalidNewBuiltin

luarmr committed:
noStringCaseMismatch

luarmr committed:
noNonoctalDecimalEscape

luarmr committed:
noSwitchDeclarations

luarmr committed:
noUnnecessaryContinue

luarmr committed:
noUnusedLabels

luarmr committed:
noUnusedPrivateClassMembers

luarmr committed:
noUnusedVariables

luarmr committed:
noVoidElementsWithChildren

luarmr committed:
noShoutyConstants

luarmr committed:
useIsNan

luarmr committed:
noUselessElse

luarmr committed:
noUnusedTemplateLiteral

luarmr committed:
noVar

luarmr committed:
useBlockStatements

luarmr committed:
useConsistentArrayType

luarmr committed:
useExponentiationOperator

luarmr committed:
useFragmentSyntax

luarmr committed:
useNumericLiterals

luarmr committed:
useNumberNamespace

luarmr committed:
useShorthandArrayType

luarmr committed:
useSelfClosingElements

luarmr committed:
useShorthandAssign

luarmr committed:
useSingleCaseStatement

luarmr committed:
useSingleVarDeclarator

luarmr committed:
useTemplate

luarmr committed:
noApproximativeNumericConstant

luarmr committed:
noCommentText

luarmr committed:
noDebugger

luarmr committed:
noConsoleLog

luarmr committed:
noDoubleEquals

luarmr committed:
noDuplicateObjectKeys

luarmr committed:
noGlobalIsFinite

luarmr committed:
noMisrefactoredShorthandAssign

luarmr committed:
noUnsafeNegation

luarmr committed:
noSparseArray

luarmr committed:
useIsArray

luarmr committed:
noFocusedTests

luarmr committed:
noSkippedTests

luarmr committed:
useValidTypeof

luarmr committed:
noConsole

luarmr committed:
useSortedClasses

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8854675039
